redis/cache.py

```diff
@@ -0,0 +1,401 @@
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+from enum import Enum
+from typing import Any, List, Optional, Union
+
+
+class CacheEntryStatus(Enum):
+    VALID = "VALID"
+    IN_PROGRESS = "IN_PROGRESS"
+
+
+class EvictionPolicyType(Enum):
+    time_based = "time_based"
+    frequency_based = "frequency_based"
+
+
+class CacheKey:
+    def __init__(self, command: str, redis_keys: tuple):
+        self.command = command
+        self.redis_keys = redis_keys
+
+    def get_redis_keys(self) -> tuple:
+        return self.redis_keys
+
+    def __hash__(self):
+        return hash((self.command, self.redis_keys))
+
+    def __eq__(self, other):
+        return hash(self) == hash(other)
+
+
+class CacheEntry:
+    def __init__(
+        self,
+        cache_key: CacheKey,
+        cache_value: bytes,
+        status: CacheEntryStatus,
+        connection_ref,
+    ):
+        self.cache_key = cache_key
+        self.cache_value = cache_value
+        self.status = status
+        self.connection_ref = connection_ref
+
+    def __hash__(self):
+        return hash(
+            (self.cache_key, self.cache_value, self.status, self.connection_ref)
+        )
+
+    def __eq__(self, other):
+        return hash(self) == hash(other)
+
+
+class EvictionPolicyInterface(ABC):
+    @property
+    @abstractmethod
+    def cache(self):
+        pass
+
+    @cache.setter
+    def cache(self, value):
+        pass
+
+    @property
+    @abstractmethod
+    def type(self) -> EvictionPolicyType:
+        pass
+
+    @abstractmethod
+    def evict_next(self) -> CacheKey:
+        pass
+
+    @abstractmethod
+    def evict_many(self, count: int) -> List[CacheKey]:
+        pass
+
+    @abstractmethod
+    def touch(self, cache_key: CacheKey) -> None:
+        pass
+
+
+class CacheConfigurationInterface(ABC):
+    @abstractmethod
+    def get_cache_class(self):
+        pass
+
+    @abstractmethod
+    def get_max_size(self) -> int:
+        pass
+
+    @abstractmethod
+    def get_eviction_policy(self):
+        pass
+
+    @abstractmethod
+    def is_exceeds_max_size(self, count: int) -> bool:
+        pass
+
+    @abstractmethod
+    def is_allowed_to_cache(self, command: str) -> bool:
+        pass
+
+
+class CacheInterface(ABC):
+    @abstractmethod
+    def get_collection(self) -> OrderedDict:
+        pass
+
+    @abstractmethod
+    def get_config(self) -> CacheConfigurationInterface:
+        pass
+
+    @abstractmethod
+    def get_eviction_policy(self) -> EvictionPolicyInterface:
+        pass
+
+    @abstractmethod
+    def get_size(self) -> int:
+        pass
```

Comment:
A more Python idiomatic approach here is to use properties instead of getters like this:
```
    @property
    @abstractmethod
    def size(self) -> int:
        pass
```
You mixed both approaches in this module, so it makes sense to unify it.

Reply:
Updated across for all getters in the class